src/xbot_monitoring.cpp

```diff
@@ -97,15 +120,11 @@ class MqttCallback : public mqtt::callback {
             }
         } else if(ptr->get_topic() == this->mqtt_topic_prefix + "action") {
             ROS_INFO_STREAM("Got action: " + ptr->get_payload());
-            std_msgs::String action_msg;
-            action_msg.data = ptr->get_payload_str();
-            action_pub.publish(action_msg);
+            action_pub.publish(parse_action_request(ptr->get_payload_str()));
         } else if(ptr->get_topic() == this->mqtt_topic_prefix + "/action") {
             // BEGIN: Deprecated code (2/2)
             ROS_WARN_STREAM("Got action on deprecated topic! Change your topic names!: " + ptr->get_payload());
-            std_msgs::String action_msg;
-            action_msg.data = ptr->get_payload_str();
-            action_pub.publish(action_msg);
+            action_pub.publish(parse_action_request(ptr->get_payload_str()));
             // END: Deprecated code (2/2)
```

Comment:
As soon as we change the wire format old apps will break anyways, so we can drop the deprecated code here, WDYT @rovo89

Reply:
My intention was to stay compatible with the old format, and then I don't see a necessity to break clients (yet).